services/web/server/tests/unit/with_dbs/10/meta/test_meta_iterations.py

```diff
@@ -0,0 +1,232 @@
+        side_effect=_mock_start,
+    )
+    # ----
+
+    resp = await client.post(
+        f"/v0/computation/pipeline/{project_uuid}:start",
+        json=RUN_PROJECT.request_payload,
+    )
+    data, _ = await assert_status(resp, web.HTTPCreated)
+    assert project_uuid == data["pipeline_id"]
+    ref_ids = data["ref_ids"]
+    assert len(ref_ids) == 3
+
+    # TODO: check: has auto-commited
+    # TODO: check: has iterations as branches
+    # TODO: retrieve results of iter1
+
+    # GET iterations ----------------------------------------------------------
+    resp = await client.get(f"/v0/repos/projects/{project_uuid}/checkpoints/HEAD")
+    body = await resp.json()
+    head_ref_id = body["data"]["id"]
+
+    assert head_ref_id == 1
+
+    resp = await client.get(
+        f"/v0/projects/{project_uuid}/checkpoint/{head_ref_id}/iterations?offset=0"
+    )
+    body = await resp.json()
+    first_iterlist = Page[ProjectIterationAsItem].parse_obj(body).data
+
+    assert len(first_iterlist) == 3
+
+    # GET workcopy project for iter 0 ----------------------------------------------
+    async def _mock_catalog_get(app, user_id, product_name, only_key_versions):
+        return [
+            {"key": s["key"], "version": s["version"]}
+            for _, s in project_data["workbench"].items()
+        ] + [{"key": "simcore/services/frontend/parameter/integer", "version": "1.0.0"}]
+
+    mocker.patch(
+        "simcore_service_webserver.catalog.get_services_for_user_in_product",
+        side_effect=_mock_catalog_get,
+    )
+
+    # extract outputs
+    for i, prj_iter in enumerate(first_iterlist):
+        resp = await client.get(prj_iter.workcopy_project_url.path)
+        assert resp.status == HTTPStatus.OK
+
+        body = await resp.json()
+        project_iter0 = body["data"]
+
+        outputs = {}
+        for nid, node in project_iter0["workbench"].items():
+            if out := node.get("outputs"):
+                outputs[nid] = out
+
+        assert len(outputs) == 1
+        assert outputs["fc9208d9-1a0a-430c-9951-9feaf1de3368"]["out_1"] == i
+
+    # ----------------------------------------------
+
+    # GET project and MODIFY iterator values----------------------------------------------
+    #  - Change iterations from 0:4 -> HEAD+1
+    resp = await client.get(f"/v0/projects/{project_uuid}")
+    assert resp.status == HTTPStatus.OK, await resp.text()
+    body = await resp.json()
+
+    # TODO: updating a project fields can be daunting because
+    # it combines nested field attributes with dicts and from the
+    # json you cannot distinguish easily what-is-what automatically
+    # Dict keys are usually some sort of identifier, typically a UUID or
+    # and index but nothing prevents a dict from user other type of key types
+    #
+    project = Project.parse_obj(body["data"])
+    new_project = project.copy(
+        update={
+            # FIXME: HACK to overcome export from None -> string
```

Comment:
still relevant?

Reply:
As I explained above, this is a hack that I was force to do until the Projects model inconsistencies are resolved. The ``models_library.utils.pydantic_models_factory`` module should be helping with this. For hte moment, I need to leave this hack in place until the aforementioned model is corrected.